### PR TITLE
raftstore: trigger split check when create new region

### PR DESCRIPTION
**Pull request by BusyJay**
Currently when a big region is split, the new region will not be checked until size diff is met. This may cause the last region too large when write a bunch of data sequentially.

@ngaut @siddontang @disksing @zhangjinpeng1987 PTAL

### CODE REVIEW

File: src/raftstore/store/peer.rs [hunk 1]
```diff
@@ -248,7 +248,7 @@ impl Peer {
             pending_cmds: Default::default(),
             peer_cache: store.peer_cache(),
             coprocessor_host: CoprocessorHost::new(),
-            size_diff_hint: 0,
+            size_diff_hint: cfg.region_check_size_diff,
```
Comment (siddontang) [inline src/raftstore/store/peer.rs]: here means all the region will be checked when the store starts, is it necessary?

Comment (BusyJay) [inline src/raftstore/store/peer.rs]: Yes. For example, a large region failed to split before shutdown. When restart, it won't check split until there is enough write operation.

File: tests/raftstore/test_split_region.rs [hunk 2]
```diff
@@ -434,3 +436,52 @@ fn test_server_split_with_stale_peer() {
...
+
+    let mut range = 1..;
+
+    cluster.run();
+
+    let pd_client = cluster.pd_client.clone();
+
+    put_till_size(cluster, region_max_size * 10, &mut range);
+
+    let min_region_cnt = (region_max_size * 10 - region_max_size) / region_split_size + 2;
```
Comment (siddontang) [inline tests/raftstore/test_split_region.rs]: region_max_size * 10 - region_max_size = region_max_size * 9 ?

Comment (BusyJay) [inline tests/raftstore/test_split_region.rs]: Yes. Just write `region_max_size * 9` may not be clear.

Comment (siddontang) [inline tests/raftstore/test_split_region.rs]: but I don't know why 10 - 1 too.

Comment (BusyJay) [inline tests/raftstore/test_split_region.rs]: Peer will split when size of region meet region_max_size, so assume the last region_max_size of data is not involved in split, there will be at least `(region_max_size * 10 - region_max_size) / region_split_size` regions. But region_max_size of data should be split too, so there will be at least 2 more regions. So the region count should not be less than `(region_max_size * 10 - region_max_size) / region_split_size + 2`.

Comment (siddontang) [inline tests/raftstore/test_split_region.rs]: you can add this comment to the code.

File: src/raftstore/store/peer.rs (hunk 1 above)
Comment (zhangjinpeng1987) [inline src/raftstore/store/peer.rs]: scan all regions will definitly raise high IO load and last for a while when store contain many GB datas when start, and this make sense to user and system

**Comment by siddontang** [comment]
LGTM

**Comment by BusyJay** [comment]
PTAL

**Comment by zhangjinpeng1987** [comment]
LGTM

**Pull request merged**